commons-profiler-api/src/main/java/ru/fix/commons/profiler/impl/ProfilerReporterImpl.java

```diff
@@ -121,7 +166,45 @@ private ProfilerCallReport buildReportAndReset(String name, SharedCounters count
                 .setReportingTime(elapsed)
 
                 .setMaxThroughputPerSecond(counters.getMaxThroughput().getMaxAndReset())
-                .setMaxPayloadThroughputPerSecond(counters.getMaxPayloadThroughput().getMaxAndReset());
+                .setMaxPayloadThroughputPerSecond(counters.getMaxPayloadThroughput().getMaxAndReset())
+
+                .setActiveCallsCount(counters.getActiveCallsCounter().sum())
+                .setActiveCallsMaxLatency(activeCallsMaxLatencyAndResetActiveCalls(counters));
+    }
+
+    private long activeCallsMaxLatencyAndResetActiveCalls(SharedCounters counters) {
+        ProfiledCallImpl longestCall = resetActiveCallsAndGetLongest(counters);
+        if (longestCall == null) {
+            return 0L;
+        }
+
+        return longestCall.timeFromCallStartInMs();
+    }
+
+    private ProfiledCallImpl resetActiveCallsAndGetLongest(SharedCounters counters) {
+        if (!enableActiveCallsMaxLatency.get() && !counters.getActiveCalls().isEmpty()) {
+            // reset is not synchronized but it does not matter.
+            // At worse we'll loose a few new calls on enabling this metric
+            counters.getActiveCalls().reset();
+            return null;
+        }
+
+        final int activeCallsLimit = numberOfActiveCallsToKeepBetweenReports.get();
+        int counter[] = new int[] {0};
+
+        ProfiledCallImpl[] longest = new ProfiledCallImpl[1];
+        counters.getActiveCalls()
+                .stream()
+                .sorted(Comparator.comparingLong(ProfiledCallImpl::startTime))
+                .forEachOrdered(call -> {
+                    if (counter[0] == 0) {
+                        longest[0] = call;
+                    } else if (counter[0] >= activeCallsLimit) {
+                        counters.getActiveCalls().remove(call);
+                    }
+                    counter[0]++;
+                });
+        return longest[0];
```

Comment:
You are removing elements from source of the stream during iteration over stream.  This approach is error prone and confusing.  
I understand that current implementation of streams under the hood will create copy of source collection to sort it and this allows you safely modify source of the stream. Plus we are using concurrent set here
But I recommend you to take a look into another approach when we do not modify source of stream during iteration over it. Example to illustrate approach: 
```
          Set<ProfiledCallImpl> topLongestActiveCalls = new HashSet<>()
            counters.getActiveCalls()
            .sorted(...) 
            .limit(aciveCallsLimit)
            .forEachOrdered( call -> {
                    if(longest[0] == null){
                        longest[0] = call;
                    }
                    topLongestAcitveCalls.put(call);
                });

          for(iter = counters.getActiveCalls().iterator(); iter.hasNext()){
            call = iter.next();
            if(!topLongestActiveCalls.contains(call)){
                iter.remove()
            }
        }
        return longest[0];
```
This is suggestion to discuss. Could you please let me know what you think.

Reply:
Agreed that it's somewhat a shady practice, redone with double iteration approach.